### ISSUE TITLE
common parent

### ISSUE THREAD

**Issue by 134130** [134130/gh-cascade#3]

**Comment by koppor** [comment]
Is this asking for a magic like "git subtree" - as I tried to describe at https://blog.flupp.de/posts/git-and-squashed-prs/ ?

**Comment by 134130** [comment]
Maybe.

There are times when you need to create small PRs to reduce the pain for PR reviewers. Not only that, but in most cases, atomic commits aimed at minimum changes are recommended.

However, there are many cases where the pace of PR reviews is slower than the pace of commits. In these cases, users need to build a separate system to manage the work they were trying to do, which is easy to forget.

I wanted to create a system to manage this through the `depends on:` statement in GitHub PRs.
- https://github.com/isaacs/github/issues/959
- https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

Basically, I use Squash merges, often making each commit squash but using trunk-feature branches to keep each commit alive, and rebase merges for main.

This issue was for the rebase strategy, but not for the repository that is currently being developed.

Thank you for your interest! 😊
